tests/room_test.go

```diff
@@ -119,40 +124,133 @@ func TestUpdateRoom(t *testing.T) {
-	_, headers, err := utils.CreateTestUser(models.User{Name: "test"})
+	id, headers, err := utils.CreateTestUser(models.User{Name: "test"})
 	if err != nil {
 		t.Error(err)
 	}
 
-	method := http.MethodDelete
+	regex := fmt.Sprintf(`{"name":"test","ownerID":%s,"usersID":\[%s\]}`, id, id)
+
+	method := http.MethodPost
+	target := "/connect"
 	test := utils.TestRUD{
 		CreateRequest:        roomCreateRequest,
 		CreateRequestHeaders: headers,
 		CreateResponse:       CreateResponseRoom{},
 		SubTests: []utils.SubTest{
-			{Name: "Success", Request: utils.Request{Method: method, Headers: headers}, ResponseCode: http.StatusNoContent, ResponseBodyRegex: ``},
-			{Name: "Correctly deleted", Request: utils.Request{Method: http.MethodGet, Headers: headers}, ResponseCode: http.StatusNotFound, ResponseBodyRegex: `{"error":"Room not found"}`},
-			{Name: "Invalid ID", Request: utils.Request{Method: method, Headers: headers, Target: "abc"}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":"Invalid room ID"}`},
-			{Name: "Not found", Request: utils.Request{Method: method, Headers: headers, Target: "987654321"}, ResponseCode: http.StatusNotFound, ResponseBodyRegex: `{"error":"Room not found"}`},
+			{Name: "Success", Request: utils.Request{Method: http.MethodGet, Headers: headers}, ResponseCode: http.StatusOK, ResponseBodyRegex: regex},
+			{Name: "Connect 2nd time", Request: utils.Request{Target: target, Method: method, Headers: headers}, ResponseCode: http.StatusConflict, ResponseBodyRegex: `{"error":"User already in room"}`},
+			{Name: "Not added 2nd time", Request: utils.Request{Method: http.MethodGet, Headers: headers}, ResponseCode: http.StatusOK, ResponseBodyRegex: regex},
+			{Name: "Invalid ID", Request: utils.Request{Method: method, Headers: headers, Target: "abc" + target}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":"Invalid room ID"}`},
+			{Name: "Not found", Request: utils.Request{Method: method, Headers: headers, Target: "987654321" + target}, ResponseCode: http.StatusNotFound, ResponseBodyRegex: `{"error":"Room not found"}`},
+		},
+	}
+	test.Run(t)
+}
+
+// TestDisconnectRoom tests the DisconnectUserToRoom function.
+func TestDisconnectRoom(t *testing.T) {
+	err := utils.ResetTable(database.DB, &models.User{}, &models.Room{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Create the user that will be used to pursue the tests.
+	id, headers, err := utils.CreateTestUser(models.User{Name: "test"})
+	if err != nil {
+		t.Error(err)
+	}
+
+	regex := fmt.Sprintf(`{"name":"test","ownerID":%s,"usersID":\[%s\]}`, id, id)
+
+	method := http.MethodPost
+	target := "/disconnect"
+	test := utils.TestRUD{
+		CreateRequest:        roomCreateRequest,
+		CreateRequestHeaders: headers,
+		CreateResponse:       CreateResponseRoom{},
+		SubTests: []utils.SubTest{
+			{Name: "Assert Setup", Request: utils.Request{Method: http.MethodGet, Headers: headers}, ResponseCode: http.StatusOK, ResponseBodyRegex: regex},
```

Comment:
I don't think this should be necessary. Indirectly it tests the creation of the room and the joining. If we do think it is good practice then we should consider adding the same kind of tests to functions like TestDeleteUser.